### ISSUE TITLE
Have setup check write permissions on directories

### ISSUE THREAD

**Issue by kchapple** [openemr/openemr#2206]
It would be a beneficial enhancement to have setup check required write permissions on the documents directory and it's sub-directories, as write permissions are required for saving patient documents, and encryption routines. It would also be beneficial to develop an administrative "System Health" page that would display a table of your php and openemr settings that could potentially diagnose permissions errors, or dependency errors.

**Comment by danehrlich1** [comment]
FYI the file that is being talked about is this one: https://github.com/openemr/openemr/blob/master/setup.php

**Comment by ardianferdianto** [comment]
Hello sir, may i claim this issues?

**Comment by bradymiller** [comment]
hi @ardianferdianto ,
This issue is yours :)

**Comment by bradymiller** [comment]
hi @ardianferdianto , Just checking how things are going on this issue? -brady

**Comment by bradymiller** [comment]
Reassigning this issue to @MERCY-VIYOLA

**Comment by MERCY-VIYOLA** [comment]
hi @bradymiller, 
Could you help validate my understanding of this requirement? 
I am supposed to add write permissions to the file [https://github.com/openemr/openemr/blob/master/setup.php]

**Comment by bradymiller** [comment]
hi @MERCY-VIYOLA ,

We are just checking to ensure the documents directory and all of its subdirectories have write access in the setup.php script (via the is_writable() php function). Right now, we are only checking the documents directory (and no the subdirectories):
https://github.com/openemr/openemr/blob/master/setup.php#L154
https://github.com/openemr/openemr/blob/master/setup.php#L165
https://github.com/openemr/openemr/blob/master/setup.php#L1400

At this block of code:
https://github.com/openemr/openemr/blob/master/setup.php#L1401-L1407
Basically need to give the $tempDir == $docsDirectory a special case where it will need to recursively go through each directory and ensure it is is_writable(). For the output of the special case, it just needs to either say something like:
directory and subdirectories are ready
versus
UNABLE to open directory or subdirectory

I hope that makes sense and is enough to get you started.
-brady

**Comment by bradymiller** [comment]
completed
https://github.com/openemr/openemr/pull/2273
